cli/azd/pkg/project/service_target_staticwebapp.go

```diff
@@ -60,9 +60,23 @@ func (at *staticWebAppTarget) Package(
 ) *async.TaskWithProgress[*ServicePackageResult, ServiceProgress] {
 	return async.RunTaskWithProgress(
 		func(task *async.TaskContextWithProgress[*ServicePackageResult, ServiceProgress]) {
+			if usingSwaConfig(packageOutput) {
+				// The swa framework service does not set a packageOutput.PackagePath during package b/c the output
+				// is governed by the swa-cli.config.json file.
+				task.SetResult(packageOutput)
+				return
+			}
+
+			// packageOutput.PackagePath != "" -> This means swa framework service is not used to build/deploy and there
+			// is no a swa-cli.config.json file to govern the output.
+			// In this case, the serviceConfig.OutputPath (azure.yaml -> service -> dist) defines where is the output from
+			// the language framework service.
+			// If serviceConfig.OutputPath is not defined, azd should use the package path defined by the language framework
+			// service.
+
 			packagePath := serviceConfig.OutputPath
 			if strings.TrimSpace(packagePath) == "" {
-				packagePath = "build"
+				packagePath = packageOutput.PackagePath
```

Comment:
Discovered this issue impacting the current implementation.
When `dist` field is not set in `azure.yaml`, we were hardcoding the packagePath to `build`, which might work for some `npm` projects, but does not work for Net applications where the package output is a temp folder.